src/NuGet.Clients/NuGet.CommandLine/Commands/SignCommand.cs

```diff
@@ -0,0 +1,190 @@
+        public string CertificatePath { get; set; }
+
+        [Option(typeof(NuGetCommand), "SignCommandCertificateStoreNameDescription")]
+        public string CertificateStoreName { get; set; }
+
+        [Option(typeof(NuGetCommand), "SignCommandCertificateStoreLocationDescription")]
+        public string CertificateStoreLocation { get; set; }
+
+        [Option(typeof(NuGetCommand), "SignCommandCertificateSubjectNameDescription")]
+        public string CertificateSubjectName { get; set; }
+
+        [Option(typeof(NuGetCommand), "SignCommandCertificateFingerprintDescription")]
+        public string CertificateFingerprint { get; set; }
+
+        [Option(typeof(NuGetCommand), "SignCommandCertificatePasswordDescription")]
+        public string CertificatePassword { get; set; }
+
+        [Option(typeof(NuGetCommand), "SignCommandCryptographicServiceProviderDescription")]
+        public string CryptographicServiceProvider { get; set; }
+
+        [Option(typeof(NuGetCommand), "SignCommandKeyContainerDescription")]
+        public string KeyContainer { get; set; }
+
+        [Option(typeof(NuGetCommand), "SignCommandHashAlgorithmDescription")]
+        public string HashAlgorithm { get; set; }
+
+        [Option(typeof(NuGetCommand), "SignCommandTimestamperDescription")]
+        public string Timestamper { get; set; }
+
+        [Option(typeof(NuGetCommand), "SignCommandTimestampHashAlgorithmDescription")]
+        public string TimestampHashAlgorithm { get; set; }
+
+        [Option(typeof(NuGetCommand), "SignCommandOverwriteDescription")]
+        public bool Overwrite { get; set; }
+
+        public override Task ExecuteCommandAsync()
+        {
+            var storeName = StoreName.My;
+            var storeLocation = StoreLocation.CurrentUser;
+            var hashAlgorithm = HashAlgorithmName.SHA256;
+            var timestampHashAlgorithm = HashAlgorithmName.SHA256;
+            
+
+            // Assert mandatory argument
+            if (Arguments.Count < 1 ||
+                string.IsNullOrEmpty(Arguments[0]))
+            {
+                throw new ArgumentException(NuGetCommand.SignCommandNoPackageException);
+            }
+
+            var packagePath = Arguments[0];
+
+            // Assert options
+            if (string.IsNullOrEmpty(Timestamper))
+            {
+                throw new ArgumentException(string.Format(CultureInfo.InvariantCulture,
+                    NuGetCommand.SignCommandNoArgumentException,
+                    nameof(Timestamper)));
+            }
+            else if (string.IsNullOrEmpty(CertificatePath) &&
+                string.IsNullOrEmpty(CertificateFingerprint) &&
+                string.IsNullOrEmpty(CertificateSubjectName))
+            {
+                throw new ArgumentException(NuGetCommand.SignCommandNoCertificateException);
+            }
+            else if (!string.IsNullOrEmpty(CertificatePath) &&
```

Comment:
I think this would more clear if we had some static helper methods for these checks. 

* Verify required option exists
* Verify no invalid option combinations
* Enum parsing + verify valid result
* With a way to throw the right message for all the above

Example: https://github.com/emgarten/NuGet.CatalogReader/blob/bcda3b86fa55a914aa0b7d3af7a3c366c0aee96e/src/NuGetMirror/Common/CmdUtils.cs#L17-L70

Reply:
Using an `else if` for all of these also implies that these things are related, but they aren't. It would be the same logic if they were all separate ifs.